### DISCUSSION TITLE
how do i get all the roles of a member

### DISCUSSION THREAD

**Discussion by CunningBard** [Rapptz/discord.py#6936] [Q&A]
i have tried to find some simillar code but i coudnt find any, essential its like this

```
adminroles = ["Owner", "Moderator", "Senior Moderator"]
@client.command()
aysnc def ban(ctx, member: discord.Member, *, reason=None):

#    userroles = get.user.roles                       
             
      if any(word in userroles for word in admin roles): 
         await member.ban(reason=reason)
         if reason is not None:
             await ctx.send(f"{member} has been banned for {reason}")
         else:
             await ctx.send(f"{member} has been banned")
```

**Comment by jelni** [comment]
You should check if the user has the Ban Members permission by adding a `@commands.has_permissions(ban_members=True)` decorator (like [here](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html#discord.ext.commands.has_permissions))

**Comment by Dorukyum** [comment]
Also to get the roles of a member simply use `member.roles` as explained [here](https://discordpy.readthedocs.io/en/latest/api.html?highlight=member.roles#discord.Member.roles)

**Comment by Kev-in123** [comment]
You should probably use `@commands.has_any_role()`
example:
```py
@bot.command()
@commands.has_any_role("Owner", "Moderator", "Senior Moderator")
async def ban(ctx, member: discord.Member, *, reason=None):
        ...
```
